src/renderer/utils/createStorage.ts

```diff
@@ -33,7 +33,7 @@ export function createStorage<T>(key: string, defaultValue: T, observableOptions
 
     // remove json-file when cluster deleted
     if (clusterId !== undefined) {
-      when(() => ClusterStore.getInstance().removedClusters.has(clusterId)).then(removeFile);
+      when(() => ClusterStore.getInstance(false)?.removedClusters.has(clusterId)).then(removeFile);
```

Comment:
Why `Singleton.getInstance(strict?: boolean)` has such definition?
Isn't it simpler to always return instance or `undefined` and use this info at place of usage (throw or whatever)?

Reply:
Basically for better error messages. Basically all users of `.getInstance()` won't work without an instance. So this doubly helps to call out where they can handle it.